tests/sentry/models/test_release.py

```diff
@@ -1327,3 +1332,60 @@ def test_simple(self):
         assert Commit.objects.filter(
             id=commit2.id, organization_id=org.id, repository_id=repo.id
         ).exists()
+
+
+@region_silo_test(stable=True)
+class ReleaseProjectManagerTestCase(TransactionTestCase):
+    def setUp(self):
+        self.redis_client = get_redis_client_for_ds()
+
+    def test_custom_manager(self):
+        self.assertIsInstance(ReleaseProject.objects, ReleaseProjectModelManager)
+
+    @receivers_raise_on_send()
+    def test_post_save_signal_runs_if_dynamic_sampling_is_disabled(self):
+        project = self.create_project(name="foo")
+        release = Release.objects.create(organization_id=project.organization_id, version="42")
+
+        with patch("sentry.models.release.schedule_invalidate_project_config") as mock_task:
+            release.add_project(project)
+            assert mock_task.mock_calls == []
+
+    @receivers_raise_on_send()
+    def test_post_save_signal_runs_if_dynamic_sampling_is_enabled_and_latest_release_rule_does_not_exist(
+        self,
+    ):
+        with Feature(
+            {
+                "organizations:dynamic-sampling": True,
+            }
+        ):
+            project = self.create_project(name="foo")
+            release = Release.objects.create(organization_id=project.organization_id, version="42")
+
+            with patch("sentry.models.release.schedule_invalidate_project_config") as mock_task:
+                release.add_project(project)
+                assert mock_task.mock_calls == []
+
+    @receivers_raise_on_send()
+    def test_post_save_signal_runs_if_dynamic_sampling_is_enabled_and_latest_release_rule_exists(
+        self,
+    ):
+        with Feature(
+            {
+                "organizations:dynamic-sampling": True,
+            }
+        ):
+            project = self.create_project(name="foo")
+            release = Release.objects.create(organization_id=project.organization_id, version="42")
+
+            # We store a boosted release for this project.
+            self.redis_client.hset(
```

Comment:
why do we need this line here?
you never assert it?

Reply:
You are right, I need the assertion. It is needed to add a boosted release, but I will do that with the other class.